app/controllers/companies_controller.rb

```diff
@@ -1,2 +1,10 @@
-class CompaniesController < ApplicationController
-end
+class CompaniesController < ApplicationController  
+  before_action :authenticate_user! do
+    @user = current_user
+    @company = @user.company
+  end
+
+  def dashboard
+    @company_setting, @user_setting = @company.dashboard_resources(@user, @company)
```

Comment:
why pass company you can get company through association

Reply:
done